internal/web/cookie.go

```diff
@@ -0,0 +1,32 @@
+package web
+
+import (
+	"net/http"
+	"strconv"
+)
+
+const advancedOptionsKey = "advancedOptions"
+
+func getAdvancedOptionsFromCookie(r *http.Request) (bool, error) {
+	advancedOptionsVal := false
+	if c, err := r.Cookie(advancedOptionsKey); err == nil {
+		if b, err := strconv.ParseBool(c.Value); err != nil {
+			return false, err
+		} else {
+			advancedOptionsVal = b
+		}
+	}
+	return advancedOptionsVal, nil
+}
+
+func setAdvancedOptionsCookie(w http.ResponseWriter, advancedOptionsEnabled bool) {
+	cookie := http.Cookie{
+		Name:     advancedOptionsKey,
+		Value:    strconv.FormatBool(advancedOptionsEnabled),
+		MaxAge:   60 * 60 * 24 * 365,
+		Secure:   true,
+		HttpOnly: true,
+		SameSite: http.SameSiteLaxMode,
```

Comment:
not that it really matters, but shouldn't SameSite=Strict work as well in this case?